Error with datetime while running tests

I'm getting this error while running the tests (docstring of tz_localize elided for brevity)

```
partd/tests/test_pandas.py:110: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
partd/pandas.py:175: in deserialize
    for (h, b) in zip(headers[2:], bytes[2:])]
partd/pandas.py:175: in <listcomp>
    for (h, b) in zip(headers[2:], bytes[2:])]
partd/pandas.py:143: in block_from_header_bytes
    values = pd.DatetimeIndex(values).tz_localize('utc').tz_convert(
/usr/lib/python3.7/site-packages/pandas/core/accessor.py:91: in f
    return self._delegate_method(name, *args, **kwargs)
/usr/lib/python3.7/site-packages/pandas/core/indexes/datetimelike.py:721: in _delegate_method
    result = operator.methodcaller(name, *args, **kwargs)(self._data)

E               TypeError: Already tz-aware, use tz_convert to convert.

/usr/lib/python3.7/site-packages/pandas/core/arrays/datetimes.py:1049: TypeError
```
This is with pandas 0.24.1.
Tests run fine if I replace line 143 in pandas.py with just `values = pd.DatetimeIndex(values)`, it seems like pandas restores the timezone just fine, not sure which version it started to work.

--- a/partd/pandas.py
+++ b/partd/pandas.py
@@ -139,9 +139,7 @@ def block_from_header_bytes(header, bytes):
                                            extension_values[1],
                                            ordered=extension_values[0])
     elif extension_type == 'datetime64_tz_type':
-        tz_info = extension_values[0]
-        values = pd.DatetimeIndex(values).tz_localize('utc').tz_convert(
-            tz_info)
+        values = pd.DatetimeIndex(values)
     return make_block(values, placement=placement)
 
 

--- a/partd/pandas.py
+++ b/partd/pandas.py
@@ -119,9 +119,8 @@ def block_to_header_bytes(block):
         extension = ('categorical_type', (values.ordered, values.categories))
         values = values.codes
     elif is_datetime64tz_dtype(block):
-        # TODO: compat with older pandas?
         extension = ('datetime64_tz_type', (block.values.tzinfo,))
-        values = np.asarray(values)
+        values = values.view('i8')
     else:
         extension = ('numpy_type', ())
 
@@ -139,7 +138,8 @@ def block_from_header_bytes(header, bytes):
                                            extension_values[1],
                                            ordered=extension_values[0])
     elif extension_type == 'datetime64_tz_type':
-        values = pd.DatetimeIndex(values)
+        tz_info = extension_values[0]
+        values = pd.DatetimeIndex(values).tz_localize('utc').tz_convert(tz_info)
     return make_block(values, placement=placement)
 
 

--- a/partd/pandas.py
+++ b/partd/pandas.py
@@ -139,7 +139,8 @@ def block_from_header_bytes(header, bytes):
                                            ordered=extension_values[0])
     elif extension_type == 'datetime64_tz_type':
         tz_info = extension_values[0]
-        values = pd.DatetimeIndex(values).tz_localize('utc').tz_convert(tz_info)
+        values = pd.DatetimeIndex(values).tz_localize('utc').tz_convert(
+            tz_info)
     return make_block(values, placement=placement)
 
 

--- a/setup.py
+++ b/setup.py
@@ -15,5 +15,10 @@ setup(name='partd',
       install_requires=list(open('requirements.txt').read().strip().split('\n')),
       long_description=(open('README.rst').read() if exists('README.rst')
                         else ''),
-      extras_require = {'complete': ['numpy >= 1.9.0', 'pandas', 'pyzmq', 'blosc']},
+      extras_require={'complete': [
+          'numpy >= 1.9.0',
+          'pandas >=0.19.0',
+          'pyzmq',
+          'blosc',
+      ]},
       zip_safe=False)

